README.md

```diff
@@ -0,0 +1,33 @@
+# Operator Framework for Charms
+
+This framework is not yet stable and is subject to change, but is available
+for early testing.
+
+## Getting Started
+
+Start by creating a charm directory with at least the following files:
+
+* `lib/charm.py` (must be executable and use Python 3.6+)
+* `hooks/install` (or `hooks/start` for K8s charms) sym-linked to `../lib/charm.py`
+* `metadata.yaml`
+
+Then install the framework into the `lib/` directory using:
+
+```
+pip install -t lib/ https://github.com/canonical/operator
+```
+
+In your `lib/charm.py` file, you will need to define a class subclassed from
+`ops.charm.CharmBase` which observes and handles at least `self.on.install`
+(or `self.on.start` for K8s charms). You will then need to call the
+`ops.main.main(YourCharm)` function, passing in the charm class that you
+defined.
```

Comment:
We need something more comprehensive here, and more accurate as well. There's no need for the charm to observe either install or start event, I believe?  It would be better to have something more pragmatic and descriptive I think, along the lines of the following:

---

Your `lib/charm.py` is the entry point for your charm logic. The minimum it needs to do is define a subclass of `CharmBase` and call the framework's main function:

```
from ops.charm import CharmBase
from ops.main import main

class MyCharm(CharmBase):
        pass

if __name__ == "__main__":
        main(MyCharm)
```

This charm does nothing, though. Typically, you'll want to observe specific juju events with logic similar to this:
```
def __init__(self, *args):
        super().__init__(*args)
        self.framework.observe(self.on.start, self.on_start)

 def on_start(self, event):
        # Handle the event here.
```

Every standard event in juju may observed that way, and you can also easily define your own events in your custom types.

---

Note that I've used the full syntax for observe, because we want to teach people about the general case that they can always use before we teach them about shortcuts.